src/consensus/validator.rs

```diff
@@ -53,12 +55,33 @@ impl ShardValidator {
         }
     }
 
-    pub fn get_validator_set(&self) -> SnapchainValidatorSet {
-        self.validator_set.clone()
+    fn get_validator_set_config(&self, height: u64) -> ValidatorSetConfig {
+        let mut result = &self.validator_set[0];
+        for config in &self.validator_set {
+            if config.effective_at <= height && config.effective_at > result.effective_at {
+                result = config;
+            }
+        }
+        result.clone()
+    }
+
+    pub fn get_validator_set(&self, height: u64) -> SnapchainValidatorSet {
```

Comment:
This will be called frequently, maybe worth caching this. Maybe within the config?

Reply:
Or worth creating a new struct called ValidatorSetConfig that can convert from the config and return the cached SnapchainValidatorSet

Reply:
I created a struct to store this in, but we don't get around cloning the value easily 🤔